### ISSUE TITLE
Module: Twoslash multi-file support weird when there's no index.ts

### ISSUE THREAD

**Issue by orta** [microsoft/TypeScript-Website#728]
**Improvement Idea:**

```
// @emitDecoratorMetadata
// @experimentalDecorators
// @showEmit
// @filename: service.ts
export class Service {}

// @filename: component.ts
import type { Service } from "./service"
```

Creates these virtual files:

```
     [
        [ '', [ '// @filename: ' ] ],
        [
          'service.ts',
          [ '// @filename: service.ts', 'export class Service {}', '', '' ]
        ],
        [
          'component.ts',
          [
            '// @filename: component.ts',
            'import type { Service } from "./service"',
            '',
            'declare var decorator: any',
            '',
            '@decorator',
            'class MyComponent {',
            '  constructor(public Service: Service) {}',
            '',
            '  @decorator',
            '  method(x: this) {}',
            '}',
            ''
          ]
        ]
      ]
```

**Why:**

The file generation from a single file is too simple

**Comment by orta** [comment]
This should be fixed by #729

**Comment by orta** [comment]
It is
